coderd/database/dbtestutil/driver.go

```diff
@@ -0,0 +1,127 @@
+package dbtestutil
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+
+	"github.com/lib/pq"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/cryptorand"
+)
+
+var (
+	_ driver.Driver             = &Driver{}
+	_ database.ConnectorCreator = &Driver{}
+	_ database.DialerConnector  = &Connector{}
+)
+
+type Driver struct {
+	name        string
+	inner       driver.Driver
+	connections []driver.Conn
+	listeners   map[chan struct{}]chan struct{}
+}
+
+func Register() (*Driver, error) {
+	db, err := sql.Open("postgres", "")
+	if err != nil {
+		return nil, xerrors.Errorf("failed to open database: %w", err)
+	}
+
+	su, err := cryptorand.StringCharset(cryptorand.Alpha, 10)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to generate random string: %w", err)
+	}
+
+	d := &Driver{
+		name:      fmt.Sprintf("postgres-test-%s", su),
+		inner:     db.Driver(),
+		listeners: make(map[chan struct{}]chan struct{}),
+	}
+
+	sql.Register(d.name, d)
```

Comment:
It seems like you're doing this so that you can get a `sql.DB` to pass to the PubSub with this instrumented driver.  But, registering it and using `sql.Open()` with the name is more complex than it needs to be.

If you allow test code to directly instantiate the instrumented connector, then you can get a `sql.DB` as:

```
db := sql.ConnectDB(connector)
```

That avoids registering the driver and worrying about a unique name.